### ISSUE TITLE
How can I expand a snippet inside a snippet?

### ISSUE THREAD

**Issue by ahmedelgabri** [SirVer/ultisnips#818]
I want to do something like this, not sure how to do it?

```vim
snippet a
${1: 'a'}
endsnippet

snippet b
" I want to expand snippet a here?
${1: 'b'}
endsnippet
```

**Comment by andreyorst** [comment]
Not sure what are you asking for.

**Comment by ahmedelgabri** [comment]
@andreyorst let me try again

With the above example (slightly modified):

```vim
snippet a
${1: 'I am snippet A'}
endsnippet

snippet b
a "<-- I want to expand snippet a here when the b snippet is expanded
${1: 'I am snippet B'}
endsnippet
```

When I expand `b` I want also that the `a` snippet inside of it to be also expanded automatically. So the output of expanding `b` in this example will look like this:

```
'I am snippet A' <-- this is the result of expanding snippet a inside snippet b
'I am snippet B'
```


Does this make sense now?

**Comment by ahmedelgabri** [comment]
Thanks @andreyorst, will have a look at the `pre-expand-actions` then.

**Comment by SirVer** [comment]
There is an approach described in https://github.com/SirVer/ultisnips/blob/master/doc/examples/snippets-aliasing/README.md, which works in many cases.

Otherwise this would be a cool feature, but very complex to implement in the current design. Since this is unlikely to happen, I will close this as wontfix.
